SerialPrograms/Source/PokemonSV/Programs/AutoStory/PokemonSV_AutoStory.cpp

```diff
@@ -0,0 +1,648 @@
+        LOOP_CHECKPOINT.set_visibility(ConfigOptionState::ENABLED);
+        START_LOOP.set_visibility(ConfigOptionState::ENABLED);
+        END_LOOP.set_visibility(ConfigOptionState::ENABLED);
+    }else{
+        START_CHECKPOINT.set_visibility(ConfigOptionState::DISABLED);
+        END_CHECKPOINT.set_visibility(ConfigOptionState::DISABLED);
+        LOOP_CHECKPOINT.set_visibility(ConfigOptionState::DISABLED);
+        START_LOOP.set_visibility(ConfigOptionState::DISABLED);
+        END_LOOP.set_visibility(ConfigOptionState::DISABLED);
+    }
+
+    if (ENABLE_TEST_REALIGN){
+        REALIGN_MODE.set_visibility(ConfigOptionState::ENABLED);
+        X_REALIGN.set_visibility(ConfigOptionState::ENABLED);
+        Y_REALIGN.set_visibility(ConfigOptionState::ENABLED);
+        REALIGN_DURATION.set_visibility(ConfigOptionState::ENABLED);
+    }else{
+        REALIGN_MODE.set_visibility(ConfigOptionState::DISABLED);
+        X_REALIGN.set_visibility(ConfigOptionState::DISABLED);
+        Y_REALIGN.set_visibility(ConfigOptionState::DISABLED);
+        REALIGN_DURATION.set_visibility(ConfigOptionState::DISABLED);        
+    }
+
+    if (ENABLE_TEST_OVERWORLD_MOVE){
+        FORWARD_TICKS.set_visibility(ConfigOptionState::ENABLED);
+    }else{
+        FORWARD_TICKS.set_visibility(ConfigOptionState::DISABLED);
+    }
+
+    if (TEST_PBF_LEFT_JOYSTICK){
+        X_MOVE.set_visibility(ConfigOptionState::ENABLED);
+        Y_MOVE.set_visibility(ConfigOptionState::ENABLED);
+        HOLD_TICKS.set_visibility(ConfigOptionState::ENABLED);
+        RELEASE_TICKS.set_visibility(ConfigOptionState::ENABLED);
+    }else{
+        X_MOVE.set_visibility(ConfigOptionState::DISABLED);
+        Y_MOVE.set_visibility(ConfigOptionState::DISABLED);
+        HOLD_TICKS.set_visibility(ConfigOptionState::DISABLED);
+        RELEASE_TICKS.set_visibility(ConfigOptionState::DISABLED);      
+    }    
+
+
+}
+
+std::string AutoStory::start_segment_description(){
+    switch(STARTPOINT){
+    case StartPoint::INTRO_CUTSCENE:
+        return "Start: Intro cutscene.";
+    case StartPoint::PICK_STARTER:
+        return "Start: Finished cutscene. Adjusted settings. Standing in left side of room.";
+    case StartPoint::NEMONA_FIRST_BATTLE:
+        return "Start: Picked the starter.";
+    case StartPoint::CATCH_TUTORIAL:
+        return "Start: Battled Nemona on the beach.";
+    case StartPoint::LEGENDARY_RESCUE:
+        return "Start: Finished catch tutorial. Walked to the cliff and heard mystery cry.";
+    case StartPoint::ARVEN_FIRST_BATTLE:
+        return "Start: Saved the Legendary. Escaped from the Houndoom cave.";
+    case StartPoint::LOS_PLATOS:
+        return "Start: Battled Arven, received Legendary's Pokeball. Talked to Nemona at Lighthouse.";
+    case StartPoint::MESAGOZA_SOUTH:
+        return "Start: At Los Platos Pokecenter.";
+    default:
+        return "";        
+    }
+}
+
+std::string AutoStory::end_segment_description(){
```

Comment:
There's one of these for every segment right? How many segments are we expecting? Like 50+ of them?

I'm wondering if this should be encapsulated into the the segments themselves. Like we define an abstract class representing a segment, then we have each segment inherit it and implement all the required stuff such as these start/end messages and the actual logic.

This approach will require refactoring the segment dropdown away from EnumDropdown to something more generic like the StringSelect options which we use in places like Pokemon name dropdowns. These are built from arrays rather than compile-time lists of enums.